docs/reference-guides/block-api/block-patterns.md

```diff
@@ -126,22 +126,24 @@ The following code sample unregisters the category named 'hero':
 unregister_block_pattern_category( 'hero' );
 ```
 
-_Note:_ 
+_Note:_
 
 `unregister_block_pattern_category()` should be called from a handler attached to the init hook.
 
 ```php
 function my_plugin_unregister_my_pattern_categories() {
   unregister_block_pattern_category( ... );
 }
- 
+
 add_action( 'init', 'my_plugin_unregister_my_pattern_categories' );
 ```
 
-## Block patterns contextual to block types
+## Block patterns contextual to block types and pattern transformations
 
 It is possible to attach a block pattern to one or more block types. This adds the block pattern as an available transform for that block type.
 
+Currently pattern transformations are available for simple blocks(blocks without inner blocks). In order for a pattern to be suggested, **every selected block must have matched with a pattern's block**. If a suggested pattern has more blocks than the selected blocks, they are added `as are` from the pattern. Every pattern block that matched with a selected block will be transformed. Noting that this doesn't necessarily mean the order of the blocks will be preserved.
```

Comment:
```suggestion
Currently these transformations are available only to simple blocks (blocks without inner blocks). In order for a pattern to be suggested, **every selected block must be present in the block pattern**.
```

Personally, I'm not sure the details about how the transformation works are needed here, it felt just confusing to me here and more suited later (as the examples show). Happy if you think otherwise.